sql/src/main/java/io/crate/action/sql/SQLOperations.java

```diff
@@ -240,18 +254,27 @@ public void parse(String statementName, String query, List<DataType> paramTypes)
             }
         }
 
-        public void bind(String portalName, String statementName, List<Object> params, @Nullable FormatCodes.FormatCode[] resultFormatCodes) {
+        public void bind(String portalName,
+                         String statementName,
+                         List<Object> params,
+                         @Nullable FormatCodes.FormatCode[] resultFormatCodes) {
             LOGGER.debug("method=bind portalName={} statementName={} params={}", portalName, statementName, params);
             checkError();
 
             try {
-                Portal portal = getOrCreatePortal(portalName, statementName);
-                portal.resultFormatCodes = resultFormatCodes;
-                portal.bulkParams.add(params);
-                if (portal.analysis == null) {
+                Portal portal = getOrCreatePortal(portalName, statementName, statement);
+
+                portal.addQuery(statementName, query);
+                portal.addStatement(statementName, statement);
+                portal.addResultFormatCodes(resultFormatCodes);
+                portal.addParams(params);
+                if (portal.getAnalysis(statementName) == null) {
                     // analyze only once in a batch operation where there is
                     // Parse -> Bind -> Describe -> Execute -> Parse -> Bind -> Describe -> Execute -> ... -> Sync
```

Comment:
Shouldn't this case now be implicitly handled inside the BulkPortal implementation?